server/base/src/main/java/org/apache/accumulo/server/fs/InitVolumeChooserEnvironmentImpl.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.fs;
+
+import java.io.IOException;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.spi.common.ServiceEnvironment;
+import org.apache.accumulo.server.ServiceEnvironmentImpl;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+
+public class InitVolumeChooserEnvironmentImpl implements VolumeChooserEnvironment {
```

Comment:
Three alternatives/variations to consider (if you haven't already) instead of having this class as-is:

1. Could make this extend VolumeChooserEnvironmentImpl, and only override those methods that are necessary,
2. Could make this an anonymous inner class inside Initialize, and
3. Could move this class to the init package, since it's an init-specific class.

Some of these variations could be combined.

In any case, I think some of this comes down to personal-preference/style. It's fine as-is (other comments aside).

Reply:
Thanks for the review, @ctubbsii! I am helping Karthick figure out the changes needed for this issue and your suggestions make sense. Regarding "only override those methods that are necessary," do you mean that we wouldn't need to include the methods that throw UnsupportedOperationException? I was thinking these would be helpful to make it clear which fields have not been initialized for the INIT scope.

Reply:
I made some suggestions about passing a KeyExtent to InitVolChooserEnvImpl.  If that works out, I don't think there would be any methods throwing UnsupportedOperationException

Reply:
I think I see what you mean now. The VolumeChooserEnvironmentImpl methods are already checking for the TABLE scope and returning IllegalStateException otherwise, so they would handle the INIT scope appropriately.

Reply:
> The VolumeChooserEnvironmentImpl methods are already checking for the TABLE scope and returning IllegalStateException otherwise, so they would handle the INIT scope appropriately.

That's what I was thinking, but Keith makes a good point about the possibility of providing table information, even during the INIT scope, in case a chooser wanted to use that information (though it wouldn't be able to use it to access per-table configuration in ZK).

Initially, I was thinking that I probably would have just make this an anonymous inner-class inside `Initialize`, only overriding those methods which would require a `ServerContext` (since we don't have one yet). Those methods are currently only `getServiceEnv()` and `getFileSystem(String)`. So, an anon-inner-class that extended `VolumeChooserEnvironmentImpl` would be very short. But, a concrete class is fine, too.